tests/framework/microvms.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Define helper functions that create predefined type of resources."""
+
+from framework.artifacts import ArtifactCollection
+from framework.builder import MicrovmBuilder
+import host_tools.network as net_tools
+from conftest import _test_images_s3_bucket
+
+
+class VmInstance:
+    """A class that describes a microvm instance resources."""
+
+    def __init__(self, config, kernel, disks, ssh_key, vm):
+        """Initialize a Vm configuration based on artifacts."""
+        self._config = config
+        self._kernel = kernel
+        self._disks = disks
+        self._ssh_key = ssh_key
+        self._vm = vm
+
+    @property
+    def config(self):
+        """Return machine config artifact."""
+        return self._config
+
+    @property
+    def kernel(self):
+        """Return the kernel artifact."""
+        return self._kernel
+
+    @property
+    def disks(self):
+        """Return an array of block file paths."""
+        return self._disks
+
+    @property
+    def ssh_key(self):
+        """Return ssh key artifact linked to the root block device."""
+        return self._ssh_key
+
+    @property
+    def vm(self):
+        """Return the Microvm object intsance."""
+        return self._vm
+
+
+# VM base class to abstract away the s3 artifacts we are using
+# to specify the configuration
+# Too few public methods (1/2) (too-few-public-methods)
+# pylint: disable=R0903
+class VMBase:
+    """Base class for constructing microvms."""
+
+    @classmethod
+    def from_artifacts(cls, bin_cloner_path, config,
+                       kernel, disks, cpu_template, start=False):
+        """Spawns a new Firecracker and applies specified config."""
+        artifacts = ArtifactCollection(_test_images_s3_bucket())
+        config = artifacts.microvms(keyword=config)[0]
```

Comment:
Could you add a comment explaining why `[0]` is used?

Reply:
Added comment.